### PR TITLE
Add querying best practices

### LINKED ISSUE #578: Create best practices for querying page

### PR DESCRIPTION
**Pull request by katarinasupe**
### Description

New page for querying best practices. We noticed that the performance optimization page holds some of the best practices, but not all. Hence, we decided to merge those two into Best practices. 
### Pull request type

Please check what kind of PR this is:

- [x] New documentation page

### Related PRs and issues

Closes #578 

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

File: pages/querying/best-practices.mdx [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+traverses through entire dataset which can lead to slower performance on a larger 
+scale datasets.
+
+<Callout> 
+**Limit query memory usage** 
+
+It is a good practice to limit [query memory
+usage](/fundamentals/storage-memory-usage#control-query-memory-usage) before
+running deep path traversals, to avoid large memory consumption for unoptimized
+queries. Keep in mind that query execution memory monotonically grows in size
```
Comment (antejavor) [inline pages/querying/best-practices.mdx:616]: monotonically is strange word for me, but maybe it is just me.

**Review by antejavor — APPROVED** [review]
LGTM

File: pages/querying/best-practices.mdx (hunk 1 above)
Comment (katarinasupe) [reply pages/querying/best-practices.mdx:616]: I agree, I just c/p that, I will update it. Thanks

**Review by kgolubic — APPROVED** [review]
Everything looks OK.

**Pull request merged**